SCP files -> Seurat object example code (SCP-3631)

jlchang opened the pull request:
Zendesk inquiry #182061 requested suggestions on how to use SCP files with Seurat.

The Rmd file is an R Studio Notebook with the example code. The .nb.html document is the code plus output from running the code. I've placed the a copy of .nb.html at https://personal.broadinstitute.org/jlchang/ for easier review (Github just renders the html as text, which is not very readable. The Rmd is human-readable but doesn't provide the output to give a non-Seurat-familiar reviewer context for what the commands are doing)

This PR fulfills SCP-3631

jlchang committed:
example code to create a Seurat object from SCP files

jlchang committed:
adding symlink, for a viewable notebook instance

jlchang committed:
workaround to view file in github failed, removing symlink

eweitz commented on scripts/object_conversion/seurat/SCP2Seurat.Rmd:
```diff
@@ -0,0 +1,109 @@
+
+Seurat's Read10X function expects the three files that comprise an Feature-Barcode Matrix to have specific names and format, all located in one directory.  For compatibility with this example, these files would be put in a directory called "10x_naming" that is located in the current working directory.
+
+To create files with the expected name and format, the following file transformations were performed for SCP1219 study files:  
```
```suggestion
To create files with the expected name and format, the following file transformations were performed for [SCP1219](https://singlecell.broadinstitute.org/single_cell/study/SCP1219/columbia-university-nyp-covid-19-lung-atlas) study files:  
```

eweitz commented on scripts/object_conversion/seurat/SCP2Seurat.Rmd:
```diff
@@ -0,0 +1,109 @@
+output: html_notebook
+---
+
+Seurat's Read10X function expects the three files that comprise an Feature-Barcode Matrix to have specific names and format, all located in one directory.  For compatibility with this example, these files would be put in a directory called "10x_naming" that is located in the current working directory.
```
```suggestion
Seurat's Read10X function expects the three files that comprise a feature-barcode matrix to have specific names and format, all located in one directory.  For compatibility with this example, these files would be put in a directory called "10x_naming" that is located in the current working directory.
```

eweitz reviewed (approved):
Looks good!  I noted a few small issues.

jlchang committed:
test lib dep change to address CI build fail

jlchang committed:
upgrade pip for CI build

jlchang committed:
revert deps list to limit tested fix to pip upgrade

jlchang committed:
Update scripts/object_conversion/seurat/SCP2Seurat.Rmd

Co-authored-by: Eric Weitz <eweitz@broadinstitute.org>

jlchang committed:
Update scripts/object_conversion/seurat/SCP2Seurat.Rmd

Co-authored-by: Eric Weitz <eweitz@broadinstitute.org>

jlchang committed:
pinning numpy and numba to try and address build issue

jlchang committed:
pin numba and llvmlite to speed pip context solving

jlchang committed:
reverting CircleCI config to not upgrade pip

jlchang committed:
update valid test input and use ingest v1.11.0

jlchang committed:
need to fix validata_metadata_file

previous functions no longer used/valid

stuck because conforms_to_metadata_convention is IngestPipeline method

bistline reviewed (approved):
Approving ahead of [216](https://github.com/broadinstitute/scp-ingest-pipeline/pull/216) being merged in scp-ingest-pipeline (and requirements.txt being updated in here).

jlchang committed:
first step to adding back metadata validation (conforms)

jlchang committed:
update tests so assessment matches refactored ingest

error used to cause script to exit, now returns False if error found

jlchang committed:
will need ingest with Exact Synonym change for current tests to pass

devonbush commented on scripts/manage_study.py:
```diff
@@ -220,10 +221,11 @@ def validate_metadata_file(parsed_args, connection):
             if verbose:
                 print(f"Retreieved file for latest metdata convention")
             convention = convention_res["response"].json()
+            validate_against_convention = True
```
so does this mean if a user needs to upload non-convention compliant data, they can't use manage study? (or if they do, they have to manually alter this line of code)?  That's totally fine, since reading feature flags is outside the scope of this PR, just want to make sure I know if it comes up on a support ticket

jlchang replied:
Because metadata validation is now required, it made sense to make validation the default for the CLI (which it wasn't). You're right that means they'd have to alter the code to do non-conventional metadata (for now).
